modules/lvgl/input/lvgl_pointer_input.c

```diff
@@ -46,62 +54,56 @@ static void lvgl_pointer_process_event(const struct device *dev, struct input_ev
 		return;
 	}
 
-	/* adjust coordinates */
-	if (cfg->swap_xy) {
-		lv_coord_t tmp;
-
-		tmp = point->x;
-		point->x = point->y;
-		point->y = tmp;
-	}
+	/* prevent re-inversion of the coordinates on touch release */
+	if (data->pending_event.state == LV_INDEV_STATE_PR) {
+		if (cfg->invert_x) {
+			if (cap->current_orientation == DISPLAY_ORIENTATION_NORMAL ||
+			    cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_180) {
+				point->x = cap->x_resolution - point->x;
+			} else {
+				point->x = cap->y_resolution - point->x;
+			}
+		}
 
-	if (cfg->invert_x) {
-		if (cap->current_orientation == DISPLAY_ORIENTATION_NORMAL ||
-		    cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_180) {
-			point->x = cap->x_resolution - point->x;
-		} else {
-			point->x = cap->y_resolution - point->x;
+		if (cfg->invert_y) {
+			if (cap->current_orientation == DISPLAY_ORIENTATION_NORMAL ||
+			    cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_180) {
+				point->y = cap->y_resolution - point->y;
+			} else {
+				point->y = cap->x_resolution - point->y;
+			}
 		}
-	}
 
-	if (cfg->invert_y) {
-		if (cap->current_orientation == DISPLAY_ORIENTATION_NORMAL ||
-		    cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_180) {
+		/* rotate touch point to match display rotation */
+		if (cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_90) {
```

Comment:
ok but now wouldn't this keep swapping x and y in case of point movement? like subsequent events that stays in "LV_INDEV_STATE_PR"?

Maybe the best thing would be to store the raw x and y in `lvgl_common_input_data` and then always do the processing starting from the store data, then they would always be swapped from the stored one into the pair in `point` rather than being swapped from the previous swap.